src/main/java/org/zalando/nakadi/controller/PartitionsController.java

```diff
@@ -33,31 +31,27 @@
 import org.zalando.nakadi.view.Cursor;
 import org.zalando.nakadi.view.CursorLag;
 import org.zalando.nakadi.view.EventTypePartitionView;
-import org.zalando.problem.MoreStatus;
 import org.zalando.problem.Problem;
-import org.zalando.problem.spring.web.advice.Responses;
 
 import javax.annotation.Nullable;
-import javax.ws.rs.core.Response;
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
-import static javax.ws.rs.core.Response.Status.NOT_FOUND;
-import static javax.ws.rs.core.Response.Status.SERVICE_UNAVAILABLE;
 import static org.springframework.http.ResponseEntity.ok;
-import static org.zalando.problem.spring.web.advice.Responses.create;
+import static org.zalando.problem.Status.NOT_FOUND;
+import static org.zalando.problem.Status.SERVICE_UNAVAILABLE;
+import static org.zalando.problem.Status.UNPROCESSABLE_ENTITY;
 
 @RestController
-public class PartitionsController {
+public class PartitionsController extends NakadiProblemControllerAdvice {
```

Comment:
@adyach I think @lmontrieux extended this class because it defines the method `create` used to generate the `Problem` response.

We would have to write some "catch/rethrow" here if we are not extending such class. What do you think? I prefer catch/rethrow instead of inheritance.

Reply:
I think spring wants you to use runtime exceptions everywhere, keep controller code linear without error handling and catch all in the controller advisor.

Reply:
Now I have a global `@ControllerAdvice` for the handlers that relate to all controllers, and a few `@ControllerAdvice` with higher priority, for those controller-specific cases. Once we merge the previous PR on this work (moving problem creation to controller), some more controller-specific handlers will also more to the specific `@ControllerAdvice` classes (such as `StorageNotFoundException`)